### PR TITLE
BFD-1519  - map revenue center unit count to v2

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-1519](https://jira.cms.gov/browse/BFD-1519)

**User Story or Bug Summary:**

User Story:

As a peering partner customer, I want Revenue Center Unit Count mapped in V2, so I can use that field.
AC:

Map claimline.unitCount to eob.item.extension
see extension below
 

Notes / References: 

From Karl - Generally, don't map two disparate value types into the same value the way we're doing here (drug code quantity OR unit count needs to be mapped to eob.item.quantity with how this is implemented).
Since we already mapped drug code to eob.item.quantity in v2 and we can't cause a backwards-incompatible change, we should solve this by making a new extension and assigning claimLine.getUnitCount to that and point people to that, so we can always have specific places that the two types of data can go. This should be part of the communications with the change.
"item" : [ {
"extension" : [ {
"url" : "https://bluebutton.cms.gov/resources/variables/rev_cntr_unit_cnt",
"valueQuantity" :

{ "value" : 1 }
},

---

### What Does This PR Do?

User Story:

As a peering partner customer, I want Revenue Center Unit Count mapped in V2, so I can use that field.
AC:

Map claimline.unitCount to eob.item.extension
see extension below
 

).
Since we already mapped drug code to eob.item.quantity in v2 and we can't cause a backwards-incompatible change, we should solve this by making a new extension and assigning claimLine.getUnitCount to that and point people to that, so we can always have specific places that the two types of data can go. This should be part of the communications with the change.
"item" : [ {
"extension" : [ {
"url" : "https://bluebutton.cms.gov/resources/variables/rev_cntr_unit_cnt",
"valueQuantity" :

{ "value" : 1 }
},

### What Should Reviewers Watch For?

Make sure all tests are covered and that the code does put the unit count in the item extension.

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?  This data already is transmitted in V1, we just added it to V2 as well.  Jason Ashbaugh gave the thumbs up that this is fine
      * [X] Yes
      * [ ] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X]I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java [hunk 1]
```diff
@@ -907,6 +907,29 @@ static void assertExtensionQuantityEquals(
       assertQuantityEquals(expectedValue.get(), (Quantity) extensionForUrl.get().getValue());
   }
 
+  /**
+   * @param ccwVariable the expected {@link CcwCodebookInterface}
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java:911]: missing java doc method description

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java [hunk 2]
```diff
@@ -225,6 +225,7 @@ private static ExplanationOfBenefit transformClaim(HHAClaim claimGroup) {
       // REV_CNTR_NCVRD_CHRG_AMT    => ExplanationOfBenefit.item.adjudication
       // REV_CNTR_NDC_QTY           => ExplanationOfBenefit.item.quantity
       // REV_CNTR_NDC_QTY_QLFR_CD   => ExplanationOfBenefit.modifier
+      // REV_CNTR_UNIT_CNT  => ExplanationOfBenefit.item.extension.valueQuantity
```
Comment (christrotteradhoc) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:228]: minor nit: this is not lined up here or in the other places below

**Review by christrotteradhoc — COMMENTED** [review]
Some very minor things noted inline. Nice work!

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java (hunk 1 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java:911]: done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java (hunk 2 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java]: done

**Review by christrotteradhoc — APPROVED** [review]
Nice work

**Comment by christrotteradhoc** [comment]
I had failed to read the description so I think the code is good to go but I'm not sure whether the api changelog adequately covers the concerns from @karlmdavis described in the description. @bfausett77 would you mind giving the changelog a read through and making any suggestions that you feel are appropriate?

**Comment by christrotteradhoc** [comment]
One other thing - since we have checked the box saying that this transmits new data I believe by process we should get the OK from @StewGoin

**Review by mackec1 — APPROVED** [review]

File: apps/bfd-server/dev/api-changelog.md [hunk 3]
```diff
@@ -1,5 +1,19 @@
 # API Changelog
 
+## BFD-1519: Map Revenue Center Unit Count in V2
+
+This extension was not previously available in the v2 claims for revenue center unit count. The
+extension is now available in the extensions for Inpatient, Outpatient, HHA, Hospice, and SNF claims.
```
Comment (bfausett77) [inline apps/bfd-server/dev/api-changelog.md:6]: I would change this last sentence to say: "The following claim types will have this extension available: Inpatient, Outpatient, HHA, Hospice, and SNF."

**Review by bfausett77 — COMMENTED** [review]
Looks good minus one requested change

File: apps/bfd-server/dev/api-changelog.md (hunk 3 above)
Comment (cbrunefearless) [reply apps/bfd-server/dev/api-changelog.md]: done

**Review by bfausett77 — APPROVED** [review]
LGTM

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**